File: DaggerDemoSimple/app/src/main/java/kynvfhn/fsoft/daggerdemosimple/MainActivity.java
package kynvfhn.fsoft.daggerdemosimple;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;

import kynvfhn.fsoft.daggerdemosimple.component.DaggerVehicleComponent;
import kynvfhn.fsoft.daggerdemosimple.component.VehicleComponent;
import kynvfhn.fsoft.daggerdemosimple.databinding.ActivityMainBinding;
import kynvfhn.fsoft.daggerdemosimple.model.Vehicle;
import kynvfhn.fsoft.daggerdemosimple.module.VehicleModule;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;
    private Vehicle vehicle;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        VehicleComponent component = DaggerVehicleComponent.builder().vehicleModule(new VehicleModule()).build();
        vehicle = component.provideVehicle();

        binding.btnSave.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                binding.txtSpeed.setText(String.valueOf(vehicle.getSpeed()));
            }
        });

    }
}

File: SharingDagger/app/src/main/java/kynvfhn/fsoft/sharingdagger/main/MainActivity.kt
package kynvfhn.fsoft.sharingdagger.main

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.provider.Settings
import kynvfhn.fsoft.sharingdagger.MyApplication
import kynvfhn.fsoft.sharingdagger.databinding.ActivityMainBinding
import kynvfhn.fsoft.sharingdagger.login.LoginActivity
import kynvfhn.fsoft.sharingdagger.registration.RegistrationActivity
import kynvfhn.fsoft.sharingdagger.settings.SettingsActivity

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private lateinit var mainViewModel: MainViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)

        val userManager = ((application as MyApplication).userManager)
        if(!userManager.isUserLoggerIn()){
            if(!userManager.isUserRegistered()){
                startActivity(Intent(this,RegistrationActivity::class.java))
                finish()
            }else{
                startActivity(Intent(this,LoginActivity::class.java))
                finish()
            }
        }else{
            setContentView(binding.root)
            mainViewModel=MainViewModel(userManager.userDataRepository!!)
            setupViews()
        }

    }

    private fun setupViews() {
        binding.hello.text = mainViewModel.welcomeText

        binding.settings.setOnClickListener {
            startActivity(Intent(this,SettingsActivity::class.java))
        }
    }


    override fun onResume() {
        super.onResume()

        binding.notifications.setOnClickListener {

        }
    }
}